### ISSUE TITLE
mask rcnn doesnt work anymore

### ISSUE THREAD

**Issue by petoor** [facebookresearch/d2go#108]
Full reproducible code :

```
from d2go.model_zoo import model_zoo
from d2go.utils.demo_predictor import DemoPredictor
from detectron2.utils.visualizer import Visualizer
from detectron2.data import MetadataCatalog, DatasetCatalog
from d2go.runner import create_runner
from detectron2.checkpoint import DetectionCheckpointer
import cv2
import matplotlib.pyplot as plt

class Instance:
    def __init__(self):
        cfg = model_zoo.get_config("mask_rcnn_fbnetv3a_dsmask_C4.yaml", trained=True)
        cfg.MODEL.DEVICE = "cpu"
        cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.70
        runner="d2go.runner.GeneralizedRCNNRunner"
        runner = create_runner(runner)
        self.model = runner.build_model(cfg)
        DetectionCheckpointer(self.model).load(cfg.MODEL.WEIGHTS)
        self.model.cpu()
        self.model.eval()
        self.predictor = DemoPredictor(self.model)
    
    def detect(self, img):
        outputs = self.predictor(img)
        v = Visualizer(img, MetadataCatalog.get("coco_2017_train"))
        out = v.draw_instance_predictions(outputs["instances"].to("cpu"))
        return out.get_image()


if __name__ == "__main__":
    #!wget http://images.cocodataset.org/val2017/000000439715.jpg -q -O input.jpg
    im = cv2.imread("./input.jpg")
    pred = Instance().detect(img=im)
    plt.imshow(pred[:,:,::-1])
    plt.show()
```

The bboxes works just fine, but the mask is random. The model initlization also states.

> Some model parameters or buffers are not found in the checkpoint:
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_0.dw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_0.pw.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_0.pw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_0.pwl.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_0.pwl.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_1.dw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_1.pw.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_1.pw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_1.pwl.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_1.pwl.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_2.dw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_2.pw.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_2.pw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_2.pwl.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_2.pwl.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_3.dw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_3.pw.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_3.pw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_3.pwl.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_3.pwl.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_4.dw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_4.pw.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_4.pw.conv.weight
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_4.pwl.bn.{bias, running_mean, running_var, weight}
> roi_heads.mask_head.feature_extractor.0.fbnetv2_0_4.pwl.conv.weight
> roi_heads.mask_head.predictor.mask_fcn_logits.{bias, weight}
> The checkpoint state_dict contains keys that are not used by the model:
>   pixel_mean
>   pixel_std
>   proposal_generator.anchor_generator.cell_anchors.0
>   roi_heads.mask_head.mask_fcn1.weight
>   roi_heads.mask_head.mask_fcn1.norm.{bias, num_batches_tracked, running_mean, running_var, weight}
>   roi_heads.mask_head.mask_fcn2.weight
>   roi_heads.mask_head.mask_fcn2.norm.{bias, num_batches_tracked, running_mean, running_var, weight}
>   roi_heads.mask_head.mask_fcn3.weight
>   roi_heads.mask_head.mask_fcn3.norm.{bias, num_batches_tracked, running_mean, running_var, weight}
>   roi_heads.mask_head.mask_fcn4.weight
>   roi_heads.mask_head.mask_fcn4.norm.{bias, num_batches_tracked, running_mean, running_var, weight}
>   roi_heads.mask_head.deconv.{bias, weight}
>   roi_heads.mask_head.predictor.{bias, weight}

Which explains the missing mask head.
This is running the code on the newest git push.

**Comment by zhanghang1989** [comment]
That' model wasn't updated for 4 month. Are you using this weights `https://mobile-cv.s3-us-west-2.amazonaws.com/d2go/models/268412271/model_0499999.pth`

**Comment by petoor** [comment]
Hi @zhanghang1989 

I am on the latest push. So i should be using that model - It also downloaded the weights using this commit.

Downloading the weights and pointing to as suggested by you works.

But according to https://github.com/facebookresearch/d2go/commit/30e798a6b3e7d8b5809da7fc1e22dcfe36d361a8 it should work out of the box.

Can you try to see if you can reproduce it? Otherwise just close the issue if it works from your fresh install

**Comment by zhanghang1989** [comment]
Thanks for the feedback. Sorry, it seems a mistake. Will take a look
